Gems/EMotionFX/Code/EMotionFX/Source/Event.cpp

```diff
@@ -25,6 +36,18 @@ namespace EMotionFX
         : m_eventDatas(AZStd::move(datas))
     {
     }
+    
+    Event& Event::operator=(const Event& other) 
+    {
+        m_eventDatas = other.m_eventDatas;
+        return *this;
+    }
+
+    Event& Event::operator=(Event&& other) 
+    {
+        m_eventDatas = AZStd::move(other.m_eventDatas);
+        return *this;
+    }
```

Comment:
It looks like `m_eventDatasChangeEvent` is missing here?

To be honest we really should write some tests for this logic to ensure all the right things are getting copied/moved as expected otherwise we won't be sure this is working correctly